### PR TITLE
[14.0] edi + edi_webservice: add 'as_bytes' option for binary files

### PR DESCRIPTION
**Pull request by simahawk**
See atomic commits for details.

Pulled out of #669

### CODE REVIEW

File: edi_oca/models/edi_exchange_record.py [hunk 1]
```diff
@@ -198,13 +198,16 @@ def _set_file_content(
...
         self[field_name] = base64.b64encode(output_string)
 
-    def _get_file_content(self, field_name="exchange_file", binary=True):
+    def _get_file_content(
+        self, field_name="exchange_file", binary=True, as_bytes=False
+    ):
         """Handy method to not have to convert b64 back and forth."""
         self.ensure_one()
         if not self[field_name]:
             return ""
```
Comment (florentx) [inline edi_oca/models/edi_exchange_record.py:207]: For consistency, do we need to manage this case as well?
```suggestion
            return "" if not as_bytes else b""
```

**Review by florentx — APPROVED** [review]
See comment about corner case.
https://github.com/OCA/edi/pull/762/files#r1194767535

File: edi_oca/models/edi_exchange_record.py (hunk 1 above)
Comment (simahawk) [reply edi_oca/models/edi_exchange_record.py:207]: hmm I don't think we need it here.
This is only for sending. If you want to manipulate the file in other circumstances you can do it in custom code.

**Pull request merged**